### DISCUSSION TITLE
Can I invoke a child machine, reach final state for it and make it start on initial state when invoked again?

### DISCUSSION THREAD

**Discussion by mogarick** [statelyai/xstate#2131] [Q&A]
I'm new to xstate.
I have the next scenario:
I have TaskA, TaskB.
I want TaskA to be represented by MachineA and TaskB with MachineB
I want to call TaskB(which has an initialStateB), from TaskA and return to a specific state on TaskA depending if TaskB matters are completed, or abandoned.
When I'm on Task A, I want the user to be able to call TaskB again and expect TaskB to be on its initialSateB.

I have made MachineA to call MachineB, transitioning it to either completed or abandoned states which I declared as final. I'm using conditionals fo onDone to determine which state to get back to.
``` json
...
invoke: {
id: "machineBId",
src: MachineB,
onDone: [
          {
            target: "stateA1",
            cond: (_, event) => {
              return event.data.success === true;
            },
          },
          {
            target: "stateA2",
            cond: (_, event) => event.data.success === false,
          },
        ]
},
...
```
The first time it works but if I try a second time to call TaskB having called it once and retuned to either stateA1 or stateA2, the MachineB continues on the final state from the first call.
So I don't know if:
* there is a way to setup MachineB to reset to initial state when it reaches a final state
* there is a way to ensure every time I invoke MachineB from MachineA, MachineB starts fresh in the initialState.
* my solution is not right for the scenario. If this is the case, what would be a recommended option.

Thank you in advance for the help.

**Comment by mogarick** [comment]
It looks the problem only happens on the https://statecharts.io/inspect visualization. I tested with console.log to check the child machine is restarted to initial state on calling again. But this doesn't seems to work on the visualization.

Here is my codesandobox for anyone who can take a look and confirm if its is a bug on de visualizer on state charts.io or what's happening.

https://codesandbox.io/s/elated-sea-u5o3g?file=/index.ts

thank you advance for any help.

**Reply by davidkpiano** [reply]
Can you make an issue for this?

**Reply by mogarick** [reply]
Hi @davidkpiano,

I've just created the issue #2135 as requested.

Thank you for your attention.
